### ISSUE TITLE
Broken debug build

### ISSUE THREAD

**Issue by simogasp** [alembic/alembic#263]
Hi,

the debug build is broken because of the `-Werror` flag.
In particular there are two warnings that are generating the errors:

* `Wdeprecated` that comes from the includes of openexr where there is a deprecated use of `throw` in the function definition. Since it is a dependency the only solution is to add a `-Wno-deprecated`

* `implicit-fallthrough` that comes from alembic code  here https://github.com/alembic/alembic/blob/master/lib/Alembic/Util/SpookyV2.cpp#L117 
```c++
switch (remainder)
    {
    case 15:
    d += ((uint64_t)u.p8[14]) << 48;
    case 14:
        d += ((uint64_t)u.p8[13]) << 40;
    case 13:
        d += ((uint64_t)u.p8[12]) << 32;
    case 12:
        d += u.p32[2];
        c += u.p64[0];
        break;
    case 11:
        d += ...
```
A series of switch-case that (I guess/hope :-) ) intentionally fallthrough (ie no `break`). There is no easy way to fix this, there are ways to fix it for [gcc](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough) by annotating the code but it is not compatible with clang. There is a `[[fallthrought]]` annotation in c++17 for this purpose but obviously it requires to upgrade the minimum version of the standard.
Maybe there is another way more complaint with all compilers that I don't know of.
Again the easy way to fix this is to add a `-Wno-error=implicit-fallthrough` flag (or fixing the switch if it is a bug! :-) )

In case I'd be happy to make the PR for this.

Thanks!

PS
This only affects non-windows builds as no flags are added for windows debug build

**Comment by fabiencastan** [comment]
Due to this problem `vcpkg install alembic` fails on linux (tested on Ubuntu 18.04).

**Comment by lamiller0** [comment]
I wasn't able to add -Wno-error=implicit-fallthrough as it caused older versions of gcc to error.

Was there a slightly different solution you ended up using?

**Comment by simogasp** [comment]
According to this https://stackoverflow.com/questions/49593240/using-wno-implicit-fallthrough-on-older-versions-of-gcc the unrecognized options are just ignored if the build success, a warning about an unrecognized option otherwise.

I ran a test here on compiler explorer https://godbolt.org/z/PERjXf
until gcc 4.4.7 it works (it does not complains)
with gcc 4.1.2 it errors
versions in between are not available but it seems that it is the case since gcc 4.4
https://gcc.gnu.org/gcc-4.4/changes.html (second last point in the Caveats)

Which is the minimum version of gcc you want the code to be compatible?

One way could be to add the no-error flag only for the minimum version of gcc that supports ignoring the unsupported options (sorry for the phrasing..)
```cmake
if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.x.y )
  # add the flag
```
obviously something similar should be done by clang


Also while at it, If i may, in the code we have
```cmake
IF ("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
    ADD_DEFINITIONS(-DDEBUG=1 -UNDEBUG)
    IF (NOT WINDOWS)
        ADD_DEFINITIONS(-Wall -Werror -Wextra -Wno-unused-parameter)
    ENDIF()
ELSEIF ("${CMAKE_BUILD_TYPE}" MATCHES "Release")
    ADD_DEFINITIONS(-DNDEBUG=1 -UDEBUG)
    IF (NOT WINDOWS)
        ADD_DEFINITIONS(-O3)
    ENDIF()
ENDIF()
```
* AFAIK CMake will add the proper NDEBUG &C automatically when you define the `CMAKE_BUILD_TYPE`, so I think we can remove them.

* The same old for `-O3`, it is automatically added by cmake according to the build type and the compiler

* i think `IF (NOT WINDOWS)` would be better replace by `if(NOT MSVC)` as you can still use clang and gcc on windows.

Let me know what you think, I can make a quick PR for all this.
Thanks!

**Comment by lamiller0** [comment]
Hmmm, I was testing with gcc4.8.

I think a PR would help here, thanks!

**Comment by simogasp** [comment]
fixed by #286
